components/src/Pagination/Pagination.js

```diff
@@ -0,0 +1,139 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import theme from "../theme";
+import { Icon } from "../Icon";
+import { Flex } from "../Flex";
+import { Text } from "../Type";
+
+const getHoverBackground = (currentPage, disabled) => {
+  if (currentPage) {
+    return theme.colors.darkBlue;
+  }
+  if (disabled) {
+    return "inital";
+  }
+  return theme.colors.lightGrey;
+};
+
+const PaginationButton = styled.button(props => ({
+  fontSize: theme.fontSizes.small,
+  padding: `${theme.space.x1} ${theme.space.x2}`,
+  lineHeight: theme.lineHeights.smallTextBase,
+  display: "flex",
+  borderRadius: "4px",
+  border: `1px solid ${theme.colors.lightGrey}`,
+  color: props.disabled ? theme.colors.grey : theme.colors.black,
+  "&:not(:last-child)": {
+    marginRight: theme.space.x2
+  },
+  cursor: props.disabled ? "default" : "pointer",
+  "&:hover": {
+    background: getHoverBackground(props.currentPage, props.disabled)
+  }
+}));
+
+const PreviousButton = props => {
+  const { disabled, onClick } = props;
+  return (
+    <PaginationButton disabled={disabled} onClick={onClick}>
+      <Icon icon="leftArrow" ml="-8px" /> Previous
+    </PaginationButton>
+  );
+};
+
+PreviousButton.propTypes = {
+  disabled: PropTypes.bool,
+  onClick: PropTypes.func
+};
+
+PreviousButton.defaultProps = {
+  disabled: false,
+  onClick: null
+};
+
+const NextButton = props => {
+  const { disabled, onClick } = props;
+  return (
+    <PaginationButton disabled={disabled} onClick={onClick}>
+      Next <Icon icon="rightArrow" mr="-8px" />
+    </PaginationButton>
+  );
+};
+
+NextButton.propTypes = {
+  disabled: PropTypes.bool,
+  onClick: PropTypes.func
+};
+
+NextButton.defaultProps = {
+  disabled: false,
+  onClick: null
+};
+
+const PageNumber = styled(PaginationButton)(props => ({
+  background: props.currentPage ? theme.colors.darkBlue : "transparent",
+  color: props.currentPage ? theme.colors.whiteGrey : theme.colors.black
+}));
+
+const SEPERATOR = "...";
+
+const getPageItemstoDisplay = (totalPages, currentPage) => {
+  const pages = Array.from({ length: totalPages }, (v, k) => k + 1);
+  if (totalPages <= 5) return pages;
+  if (currentPage === 1) return [...pages.slice(currentPage - 1, 2), SEPERATOR, totalPages];
+  if (currentPage === totalPages) return [1, SEPERATOR, ...pages.slice(totalPages - 2, totalPages)];
```

Comment:
Just noticed that we might have to fix this for  currentPage <= 3..
since we should have 
[1, 2*, 3, ..., 10]
[1, 2, 3*, 4,..., 10]
but currently i think we would get 
[1, .. 1, 2*. 3,... 10]
and same for 3 numbers from the last page